Northstar.Client/mod/scripts/vscripts/ui/_menus.nut

```diff
@@ -178,6 +178,23 @@ void function UICodeCallback_ToggleInGameMenu()
 	}
 }
 
+
+void function TimeoutHandler()
+{
+	//Singleplayer "Continue" counts as finished loading and shouldn't be affected
+	printt( "TimeoutHandler has Initialized" )
+	EndSignal( uiGlobal.signalDummy, "LevelShutdown" ) // Cancels thread if map is unloaded
+
+	wait GetConVarFloat( "ns_max_load_time" )
+	if ( uiGlobal.isLoading == true ) //Disconnects client after 30 seconds of loading
+	{
+		printt( "TimeoutHandler disconnected because loading took too long" )
+		UICodeCallback_LevelLoadingFinished( true ) //Removes loading screen
+		Disconnect() // Loads menu
+		OpenErrorDialog( format( "Loading timed out (Over %i seconds)", GetConVarInt( "ns_max_load_time" ) ) ) //Opens error dialog to notify user
```

Comment:
```suggestion
		OpenErrorDialog( format( "Loading timed out after %.1f seconds.\nYou can increase the limit in the Mod Settings on the Main Menu" ) ) ) // Opens error dialog to notify user
```
This should be localized though

Reply:
I am uploading that now, worded slightly differently